File: ScoreApi/Models/Contestant.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ScoreApi.Models
{
    [Table("Contestants")]
    public class Contestant
    {
        [Key]
        public int ContestantId { get; set; }

        public int ContestantNumber { get; set; }

        [Required]
        [MaxLength(100)]
        public string FullName { get; set; } = string.Empty;

        [MaxLength(100)]
        public string Category { get; set; } = string.Empty;

        [Required]
        [MaxLength(20)]
        public string Status { get; set; } = "Active";

        // Navigation
        public ICollection<ScoreSubmission> ScoreSubmissions { get; set; } = new List<ScoreSubmission>();
    }
}

File: frontend/src/types/contestant.ts
export type ContestantStatus = 'Active' | 'Inactive' | 'Disqualified';

export interface Contestant {
    contestantId: number;
    contestantNumber: number;
    fullName: string;
    category: string;
    status: ContestantStatus;
}

export interface CreateContestantRequest {
    contestantNumber: number;
    fullName: string;
    category: string;
    status: ContestantStatus;
}

export interface UpdateStatusRequest {
    status: ContestantStatus;
}